suite/tests/api/detach_state.c

```diff
@@ -408,7 +408,94 @@ START_FILE
         push     rbp @N@\
         push     rsi @N@\
         push     rdi
-#        define POPALL        \
+/* This does NOT make xsp have its pre-push value. */
+#        ifdef __AVX512F__
+#            define PUSHALL                 \
+        lea      rsp, [rsp - NUM_SIMD_REGS*SIMD_REG_SIZE - \
+                       NUM_OPMASK_REGS*OPMASK_REG_SIZE] @N@     \
+        vmovups  [rsp +  0*SIMD_REG_SIZE], zmm0 @N@ \
+        vmovups  [rsp +  1*SIMD_REG_SIZE], zmm1 @N@ \
+        vmovups  [rsp +  2*SIMD_REG_SIZE], zmm2 @N@ \
+        vmovups  [rsp +  3*SIMD_REG_SIZE], zmm3 @N@ \
+        vmovups  [rsp +  4*SIMD_REG_SIZE], zmm4 @N@ \
+        vmovups  [rsp +  5*SIMD_REG_SIZE], zmm5 @N@ \
+        vmovups  [rsp +  6*SIMD_REG_SIZE], zmm6 @N@ \
+        vmovups  [rsp +  7*SIMD_REG_SIZE], zmm7 @N@ \
+        vmovups  [rsp +  8*SIMD_REG_SIZE], zmm8 @N@ \
```

Comment:
ifdef X64 missing?

Reply:
All of this code is inside ifdef X64 up above.  Some should be shared when 32-bit support is added, but for now that's not there.